### PR TITLE
Node version support and lazy loading changes

### PR DESCRIPTION
**Pull request by ankur-dwivedi**

### CODE REVIEW

File: src/components/IKImage/index.tsx [hunk 1]
```diff
@@ -138,25 +138,27 @@ const IKImage = (props: Omit<ImageProps, "src" | "loading" | "loader"> & IKImage
...
       if (updatedRestProps.height) delete updatedRestProps["height"];
       if (updatedRestProps.width) delete updatedRestProps["width"];
       updatedRestProps["fill"] = true;
     }
     setImageProps(updatedRestProps);
   }, []);
 
+  useEffect(()=>{
+    if(lqip?.active)
+      console.warn("Eager is applied")
```
Comment (ahnv) [inline src/components/IKImage/index.tsx]: Make the error properly descriptive and add a `[imagekitio-react]` at the start

Something like `[imagekitio-react] paramters lqip and loading cannot be used together`

and show the warning only when loading is passed as a prop

Comment (ankur-dwivedi) [reply src/components/IKImage/index.tsx]: done

File: package.json [hunk 2]
```diff
@@ -52,7 +52,7 @@
     "imagekit-javascript": "^3.0.2"
   },
   "engines": {
-    "node": ">=18.17.0"
+    "node": ">=12"
   },
   "peerDependencies": {
     "next": "^12 || ^13 || ^14",
```
Comment (ahnv) [inline package.json]: can you add next 15 here and test it out once?

Comment (ankur-dwivedi) [reply package.json]: Done i have tested and made changes to make it compatible with next 15

File: package.json [hunk 3]
```diff
@@ -52,10 +52,10 @@
...
   },
   "engines": {
-    "node": ">=18.17.0"
+    "node": ">=12"
   },
   "peerDependencies": {
-    "next": "^12 || ^13 || ^14",
+    "next": "^12 || ^13 || ^14 || ^15",
     "react": "^16.13.1 || ^17.0.0 || ^18.0.0",
     "react-dom": "^16.13.1 || ^17.0.0 || ^18.0.0"
```
Comment (ahnv) [inline package.json]: add react peer dependencies for react 19 also, then we should be good to go

Comment (ankur-dwivedi) [reply package.json]: done

**Review by ahnv — APPROVED** [review]

**Pull request merged**